components/renku_data_services/project/db.py

```diff
@@ -892,3 +897,151 @@ async def delete_session_secrets(
+            project_v1_id=project_v1_id,
+            launcher_id=result_launcher.id if result_launcher else None,
+        )
+
+        if migration_orm.project_id is None:
+            raise errors.ValidationError(message="Project ID cannot be None for the migration entry.")
+
+        session.add(migration_orm)
+        await session.flush()
+        await session.refresh(migration_orm)
+
+        return created_project
+
+    async def get_migration_by_v1_id(self, user: base_models.APIUser, v1_id: int) -> models.Project:
+        """Retrieve all migration records for a given project v1 ID."""
+        if user.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+
+        async with self.session_maker() as session:
+            stmt = select(schemas.ProjectMigrationsORM).where(schemas.ProjectMigrationsORM.project_v1_id == v1_id)
+            result = await session.execute(stmt)
+            project_ids = result.scalars().first()
+
+            if not project_ids:
+                raise errors.MissingResourceError(message=f"Migration for project v1 with id '{v1_id}' does not exist.")
+
+            # NOTE: Show only those projects that user has access to
+            allowed_projects = await self.authz.resources_with_direct_membership(user, ResourceType.project)
+            project_id_list = [project_ids.project_id]
+            stmt = select(schemas.ProjectORM)
+            stmt = stmt.where(schemas.ProjectORM.id.in_(project_id_list))
+            stmt = stmt.where(schemas.ProjectORM.id.in_(allowed_projects))
+            result = await session.execute(stmt)
+            project_orm = result.scalars().first()
+
+            if project_orm is None:
+                raise errors.MissingResourceError(
+                    message="Project migrated does not exist or you dont have permissions to open it."
+                )
+
+            return project_orm.dump()
+
+    async def get_migration_by_project_id(
+        self, user: base_models.APIUser, project_id: ULID
+    ) -> models.ProjectMigrationInfo | None:
+        """Retrieve migration info for a given project v2 ID."""
+
+        if user.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+
+        async with self.session_maker() as session:
+            stmt_project = select(schemas.ProjectORM.id).where(schemas.ProjectORM.id == project_id)
+            res_project = await session.scalar(stmt_project)
+            if not res_project:
+                raise errors.MissingResourceError(
+                    message=f"Project with ID {project_id} does not exist or you do not have access to it."
+                )
+
+            stmt = select(schemas.ProjectMigrationsORM).where(schemas.ProjectMigrationsORM.project_id == project_id)
+            result = await session.execute(stmt)
+            project_migration_orm = result.scalars().first()
```

Comment:
issue: we should probably also check that the user has access to the v2 project here

Reply:
sure, added here [bd88030](https://github.com/SwissDataScienceCenter/renku-data-services/pull/628/commits/bd88030f6c8b6f48b09d173918eca896698fd13d#diff-e6ad745574f64b4241525aaa0d39f7736545db13f97f9092f00b723257970b9c)